[feat] 이미지 업로드 기능 구현

pie2457 opened the pull request:
## Issues
- #10 

## What is this PR? 👓
- 이미지 업로드 기능을 구현했습니다.
- 이미지 업로드 기능을 테스트했습니다.

##  📖 다시 한번 정리하기
- `S3Properties` : .yml 파일에서 읽어온 accessKey, secretKey, bucket, region을 가지고 있음
- `S3Config`: S3에 이미지를 올리기 위해 사용되는 `AmazonS3Client`를 `bean`으로 등록
- `ImageFile` : multipartfile자체를 s3에 올리는게 아닌 이미지 파일로 변환시켜 s3에 저장해야 하기 때문에 multipartfile에 담긴 필요한 정보들만 분리 해 도메인 객체로 저장. 
  -  s3에 저장 시킬때 이미지의 파일 이름이 중복되면 안되므로 randomUUID를 생성 해 fileName에 붙여넣음
- `ImageService`: mulitpartfile로 들어온 데이터를 ImageFile 객체로 변환 시켜주고 imageUploader에게 업로딩 역할을 던져줌.
- `ImageUploader`: 이미지를 업로딩 하는 역할만 한다.

pie2457 committed:
docs #10 : S3Properties 작성

s3에 사진을 올리기 위해 필요한 환경변수를 .yml파일에서 읽어오도록 설정.

pie2457 committed:
feat #10 : S3Config 작성

s3에 이미지 파일을 올리기 위해 AmazonS3Client bean 등록

pie2457 committed:
docs #10 : build.gradle에 S3 관련 파일 의존성 추가

pie2457 committed:
feat #10 : 에러코드 작성

pie2457 committed:
feat #10 : 이미지 업로드 기능 구현

pie2457 committed:
test #10 : 이미지 업로드 기능 테스트

pie2457 committed:
refactor #10 : 접근제어자 및 .yml 수정

- 생성자의 접근제어자가 private이어도 될 거 같아서 변경했습니다.
- 경로 설정이 잘못된 이름으로 되어있어서 수정했습니다.

23Yong commented on src/main/java/pie/tomato/tomatomarket/domain/ImageFile.java:
```diff
@@ -0,0 +1,90 @@
+		public static String findEnum(String contentType) {
+
+			for (ImageContentType imageContentType : ImageContentType.values()) {
+				if (imageContentType.getContentType().equals(contentType.toLowerCase())) {
```
`getContentType().equalsIgnoreCase(contentType)`도 있으니 참고해주세용

pie2457 replied:
오옹 대소문자를 비교하지 않는 메서드구나 감사합니당👏🏻

23Yong commented on src/main/java/pie/tomato/tomatomarket/domain/ImageFile.java:
```diff
@@ -0,0 +1,90 @@
+		private final String contentType;
+
+		public static String findEnum(String contentType) {
+
```
한 줄 띄어져있는 것 같아용 붙여주세욤

pie2457 replied:
81라인 얘기하시는거죠???

23Yong commented on src/main/java/pie/tomato/tomatomarket/infrastructure/config/properties/S3Properties.java:
```diff
@@ -0,0 +1,39 @@
+	@Getter
+	@RequiredArgsConstructor
+	public static class S3 {
+		private final String bucket;
```
위에는 한 줄 띄었는데 여기는 한 줄 안띄어져 있네용

pie2457 replied:
몰랐어요 ! 감사합니당 !

23Yong commented on src/test/java/pie/tomato/tomatomarket/application/unit/ImageServiceTest.java:
```diff
@@ -0,0 +1,45 @@
+@SpringBootTest
+class ImageServiceTest {
+
+	@InjectMocks
+	private ImageService imageService;
+	@Mock
+	private ImageUploader imageUploader;
```
`@SpringBootTest`라 다음과 같이 해도 좋을 것 같아요
```java
@Autowired
private ImageService imageService;
@MockBean
private ImageUploader imageUploader;
```

pie2457 replied:
`@ExtendWith(MockitoExtension.Class)`을 써도 되는거죵?

23Yong reviewed (approved):
수고하셨습니당~ 깔끔하네욤

pie2457 committed:
refactor #10 : 리뷰사항 반영

- StringBootTest를 @ExtendWith 애노테이션을 사용해 테스트 변경
- ImageFile에서 불필요한 공백 제거
- S3Properties에서 컨벤션을 위한 공백 추가
